### PR TITLE
build.gradle cleanup and update of plugins

### PR DESCRIPTION
**Pull request by jutoft**
I ran a travis build, but i have of cause not set it up for releasing.
I hope you can try to trigger that yourself?

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,23 +1,12 @@
...
-apply plugin: 'groovy'
-apply plugin: 'maven-publish'
-apply plugin: 'com.jfrog.bintray'
-apply plugin: 'org.ajoberstar.github-pages'
-apply plugin: 'java-gradle-plugin'
-
-def compatibilityVersion = 1.6
-sourceCompatibility = compatibilityVersion
-targetCompatibility = compatibilityVersion
+sourceCompatibility = "1.6"
```
Comment (bmuschko) [inline build.gradle]: What's the reason for removing the target compatibility?

File: build.gradle [hunk 2]
```diff
@@ -1,23 +1,12 @@
...
-
-    dependencies {
-        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.4'
-        classpath 'org.ajoberstar:gradle-git:1.4.2'
-    }
+plugins {
+    id "groovy"
+    id "java-gradle-plugin"
+    id "com.jfrog.bintray" version "1.8.4"
+    id "maven-publish"
```
Comment (bmuschko) [inline build.gradle]: I believe that the Maven Publish plugin is already applied by the Bintray plugin if I am not mistaken.

File: build.gradle (hunk 1 above)
Comment (jutoft) [reply build.gradle]: Target is equal to source, unless you override it. So there is no reason to define it unless you want them to be different.

File: build.gradle (hunk 2 above)
Comment (jutoft) [reply build.gradle]: I will remove it but If i havent done it tomorrow then it probably will jave to wait 2 weeks

File: build.gradle (hunk 1 above)
Comment (jutoft) [reply build.gradle]: For java the options is passed directly to java as far as i can see in the code.

Check doc for javac

https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javac.html#crosscomp-options

Comment (jutoft) [reply build.gradle]: It appears to differ from java version to java version. So i guess its best to just define both.

Comment (bmuschko) [reply build.gradle]: OK, can you put the target compatibility back in place?

**Comment by bmuschko** [comment]
Judging by the failed Travis build it looks like you do need the Maven Publish plugin. Could you put it back in place? Sorry for the run-around.

**Comment by bmuschko** [comment]
Thanks for the PR. This has been merged.

**Pull request merged**